src/shims/unix/linux/fs.rs

```diff
@@ -0,0 +1,167 @@
+use rustc_middle::ty::ScalarInt;
+
+use crate::{machine::SIGRTMAX, *};
+use epoll::{Epoll, EpollEvent};
+use event::Event;
+use socketpair::SocketPair;
+
+use shims::unix::fs::EvalContextExtPrivate;
+
+pub mod epoll;
+pub mod event;
+pub mod socketpair;
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriInterpCx<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
+    fn epoll_create1(&mut self, flags: &OpTy<'tcx, Provenance>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        let flags = this.read_scalar(flags)?.to_i32()?;
+
+        let epoll_cloexec = this.eval_libc_i32("EPOLL_CLOEXEC")?;
+        // FIXME handle cloexec
+        let _fd_cloexec = this.eval_libc_i32("FD_CLOEXEC")?;
+        if flags == epoll_cloexec {
+            // FIXME set close on exec, FD_CLOEXEC
+        } else if flags != 0 {
+            let einval = this.eval_libc("EINVAL")?;
+            this.set_last_error(einval)?;
+            return Ok(-1);
```

Comment:
Usually we do `throw_unsup_format` on unsupported flags... or are these truly all flags that even exist on the real implementation?